apps/prairielearn/src/sprocs/users_select_or_insert.sql

```diff
@@ -68,7 +68,7 @@ BEGIN
             iap.institution_id = institution.id
             AND ap.name = authn_provider_name;
 
-        IF NOT FOUND THEN
+        IF NOT FOUND AND authn_provider_name != 'dev' THEN
```

Comment:
Magic strings like this are typically deserving of comments. I'd +1 Jonatan's suggestion to do this up on line 62; the `dev` authn provider doesn't appear to have a row in `authn_providers` and so the query above will _never_ find a row.